src/components/marker-chart/Canvas.js

```diff
@@ -97,40 +98,66 @@ class MarkerChartCanvas extends React.PureComponent<Props, State> {
       Math.ceil(viewportBottom / rowHeight),
       markerTimingAndBuckets.length
     );
-
+    const markerIndexToTimingRow = this._getMarkerIndexToTimingRow(
+      markerTimingAndBuckets
+    );
+    const rightClickedRow: number | void =
+      rightClickedMarkerIndex === null
+        ? undefined
+        : markerIndexToTimingRow.get(rightClickedMarkerIndex);
+    const newRow: number | void =
+      hoveredItem === null
+        ? undefined
+        : markerIndexToTimingRow.get(hoveredItem);
+        
     if (isHoveredOnlyDifferent) {
       // Only re-draw the rows that have been updated if only the hovering information
       // is different.
-      const markerIndexToTimingRow = this._getMarkerIndexToTimingRow(
-        markerTimingAndBuckets
-      );
 
       const oldRow: number | void =
         prevHoveredItem === null
           ? undefined
           : markerIndexToTimingRow.get(prevHoveredItem);
-      const newRow: number | void =
-        hoveredItem === null
-          ? undefined
-          : markerIndexToTimingRow.get(hoveredItem);
 
       if (newRow !== undefined) {
         this.clearRow(ctx, newRow);
+        this.highlightRow(ctx, newRow);
         this.drawMarkers(ctx, hoveredItem, newRow, newRow + 1);
       }
       if (oldRow !== undefined && oldRow !== newRow) {
-        this.clearRow(ctx, oldRow);
+        if (oldRow !== rightClickedRow) {
+          this.clearRow(ctx, oldRow);
+        }
```

Comment:
(no action to do, it's just a remark): it's interesting because this actually means that we didn't need to `clearRow` previously! Just redrawing the markers was enough...